src/shortcuts.cc

```diff
@@ -0,0 +1,53 @@
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonArray>
+#include <QJsonValue>
+#include <QFile>
+#include <QVariant>
+#include "shortcuts.h"
+#include "PlatformUtils.h"
+#include "printutils.h"
+#include "qtgettext.h"
+
+ShortCutConfigurator::ShortCutConfigurator(){
+
+
+}
+
+void ShortCutConfigurator::apply(const QList<QAction *> &actions)
+{
+	std::string absolutePath = PlatformUtils::userConfigPath()+"/shortcuts.json";
+
+	QString finalPath = QString::fromLocal8Bit(absolutePath.c_str());
+
+	QFile jsonFile(finalPath);
+	
+    //check if a User-Defined Shortcuts file exists or not
+	if (!jsonFile.open(QIODevice::ReadOnly | QIODevice::Text))
+	{
+	return;
+	}
+
+	QByteArray jsonData = jsonFile.readAll();
+	QJsonDocument doc = QJsonDocument::fromJson(jsonData);
+	QJsonObject jObject = doc.object();
+	QVariantMap json_map = jObject.toVariantMap();
+
+    for (auto &action : actions) 
+    {
+        const QString shortCut(action->shortcut().toString(QKeySequence::NativeText));
+        //handle the case, if shortcut is not defined for an action
+        if (shortCut.isEmpty())
```

Comment:
That's checking the default shortcut of the action, correct? Why prevent setting a shortcut for actions that have none by default? With the configuration in place we may end up removing a couple of more obscure shortcuts and leave it to the user to define something that fits their needs.

Reply:
I thought this would create a bad experience for the user. If he tries to add more shortcuts, as there will be conflicts of shortcuts and the number of shortcuts keys available are limited.
So, if you want we can change this. What would you suggest?

Reply:
Now I think you are correct @t-paul . So, now I have removed this check and the user can introduce new shortcuts. That makes sense.